### ISSUE TITLE
Please consider providing an environment variable which disables box64.

### ISSUE THREAD

**Issue by josch** [ptitSeb/box64#1237]
Thanks to binfmt support, once installed, box64 is always "active" and will be used to interpret x86_64 binaries. Sometimes this is either not desired or client applications want to figure out what tool it is providing transparent x86_64 emulation via the binfmt_misc mechanism. Other providers of binfmt interpreters solve this via an environment variable. For example qemu has QEMU_VERSION which, if set, will disable emulation but just print the qemu version string when attempting to run a foreign architecture binary via binfmt_misc.

I'm asking for a similar mechanism for box64. Please consider making it possible to

 1. disable box64 via an environment variable
 2. figure out that it is box64 doing the emulation

This is also useful for the `arch-test` tool which has a `-n` switch, disabling known binfmt_misc providers by setting the respective environment variable.

I propose the `BOX64_VERSION` environment variable which, if set, would just print the box64 version and exit.

Thanks!

**Comment by ptitSeb** [comment]
I'm not sure I understand: if `BOX64_VERSION` is set to something not 0, the program passed to box64 should be launch without box64?
How is that supposed to work? If you launch a x86_64 program, binfmt will redirect to box64, that, if BOX64_VERSION is set will redirect to launch natively, wich, with binfmt, will redirect to box64 and will loop indefently? Or am I missing something?

**Comment by josch** [comment]
No, the program should not launch at all. That's what `QEMU_VERSION` does as well. If `QEMU_VERSION` is set in the environment, then the program will *not* launch and qemu itself will just print its version string and exit.

Here is what qemu does if `QEMU_VERSION` is set:

https://gitlab.com/qemu-project/qemu/-/blob/master/linux-user/main.c?ref_type=heads#L419

**Comment by ptitSeb** [comment]
I'm really confused on what use case this could serve.

The is \BOX64_EXIT` thar just exit box64 that is define. It's only on rcfile for now but could be extended to command line too I guess

**Comment by josch** [comment]
Yes, if I could set `BOX64_EXIT=1` as an environment variable, that would solve this issue as well.

I described the use-case in my initial message. Let me rephrase:

 1. I have a tool like `arch-test` which is supposed to tell me whether architecture X can be executed or not. The tool allows me to disable binfmt misc support in user-space for known emulators like qemu or wine using environment variables. It doesn't support box64 yet, because it does not yet have such an environment variable as qemu and wine have. If box64 would gain such an environment variable, then `arch-test` could gain support for box64 as well. Currently, when having box64 installed, the output of `arch-test -n ${arch}` is incorrect because `arch-test` cannot disable box64.
 2. I have a tool which knows that if box64 is the one doing the x86_64 emulation, it will break (for example because it runs the foreign architecture binaries inside a chroot where box64 is not available and box64 is not compiled statically). That tool would like to be able to give the user a descriptive error message about the fact that right now, x86_64 is emulated box64 and that cannot work. Currently, it cannot find out whether box64 is the binfmt_misc provider.

I can propose a pull request that implements `BOX64_VERSION` equivalent to what `QEMU_VERSION` does for qemu if you like.

**Comment by ptitSeb** [comment]
You can submit a PR yes.
